skfda/preprocessing/smoothing/validation.py

```diff
@@ -54,72 +62,90 @@ def gcv(fdatagrid, s_matrix, penalisation_function=None):
-            * (1 - s_matrix.diagonal().mean()) ** -2)
+    def __init__(self, penalization_function=None):
+        self.penalization_function = penalization_function
 
+    def __call__(self, estimator, X, y):
+        y_est = estimator.transform(X)
 
-def minimise(fdatagrid, parameters,
-             smoothing_method=kernel_smoothers.nw, cv_method=gcv,
-             penalisation_function=None, **kwargs):
-    """Chooses the best smoothness parameter and performs smoothing.
+        hat_matrix = estimator.hat_matrix_
 
-    Performs the smoothing of a FDataGrid object choosing the best
-    parameter of a given list using a cross validation scoring method.
-
-    Args:
-        fdatagrid (FDataGrid): FDataGrid object.
-        parameters (list of double): List of parameters to be tested.
-        smoothing_method (Function): Function that takes a list of
-            discretised points, a parameter, an optionally a weights matrix
-            and returns a hat matrix or smoothing matrix.
-        cv_method (Function): Function that takes a matrix,
-            a smoothing matrix, and optionally a weights matrix and
-            calculates a cross validation score.
-        penalisation_function(Fuction): if gcv is selected as cv_method a
-            penalisation function can be specified through this parameter.
+        if self.penalization_function is None:
+            def penalization_function(hat_matrix):
+                return (1 - hat_matrix.diagonal().mean()) ** -2
+        else:
+            penalization_function = self.penalization_function
 
-    Returns:
-        dict: A dictionary containing the following:
+        return -(np.mean(((y.data_matrix[..., 0] - y_est.data_matrix[..., 0])
+                          / (1 - hat_matrix.diagonal())) ** 2)
+                 * penalization_function(hat_matrix))
 
-            {
-                'scores': (list of double) List of the scores for each
-                    parameter.
 
-                'best_score': (double) Minimum score.
+class SmoothingParameterSearch(GridSearchCV):
+    """Chooses the best smoothing parameter and performs smoothing.
 
-                'best_parameter': (double) Parameter that produces the
-                    lesser score.
-
-                'hat_matrix': (numpy.darray) Hat matrix built with the best
-                    parameter.
+    Performs the smoothing of a FDataGrid object choosing the best
+    parameter of a given list using a cross validation scoring method.
 
-                'fdatagrid': (FDataGrid) Smoothed FDataGrid object.
+    Note:
+        This is similar to fitting a scikit-learn GridSearchCV over the
+        data, using the cv_method as a scorer.
 
-            }
+    Args:
+        estimator (smoother estimator): scikit-learn compatible smoother.
+        param_values (iterable): iterable containing the values to test
+            for *smoothing_parameter*.
+        scoring (scoring method): scoring method used to measure the
```

Comment:
Only with our documentation I haven't been able to infer the default scorer. 
I had to look at the scikit-learn documentation to see that the estimator score method is used.

Reply:
Improved in f7f1313b51dbd283d451a205969d6d28c19dc80b.